students/brandon_nguyen/session05/mailroom3.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+# Week5 Excercise mailroom part 3 - Exception, List comprehension
+# Student: Brandon Nguyen - Au2018
+import sys
+import unittest
+from decimal import Decimal
+from datetime import datetime
+
+
+# global define data structure
+# dictionary {person:[donations]}
+donor_db = {"Brandon Nguyen": [4500.33, 350.87, 300.05],
+            "Shawn DeGraw": [5500.00, 345.05, 571.33],
+            "Jacqueline Lee": [4300.00, 3200.00, 230.13],
+            "Aidan Nguyen": [4300.00, 200.00, 238.23],
+            }
+promptText = "\n".join(("\nWelcome to the mailroom program!\n",
+                        "Please choose an options:\n\n",
+                        "1 - Send a Thank You to a single donor.",
+                        "2 - Create a Report.",
+                        "3 - Send letters to all donors.",
+                        "q - Quit",
+                        ">>> "))
+
+subMenuPrompt = ("Please Chose an option:\n\nlist - "
+                 "To display a list of current donors.\n"
+                 "1    - To enter name of donor to update.\n"
+                 "q    - To exit this menu.\n"
+                 ">>> ")              
+# building switch case menu with dictionary
+# think about a main menu function that can be called any where.
+
+
+def show_menu(promptxt, select_dict):
+    while True:
+        try:
+            menu_selected = input(promptxt)
+            if select_dict[menu_selected]() == 'Exit Menu':
+                break
+        except KeyError as err:
+            print()
+            print("You entered: {} incorrect option!".format(err))
+
+
+def send_ty():
+    show_menu(subMenuPrompt, sub_menu_dict)
+
+
+def create_rpt():
+    """
+    This function is to print the report of donor.  
+    """
+    # Currently need more work on the format as 0.0 problem not yet solved.
+    print()
+    # creating new list with sorted total in reverse already for easy printing.
+    newList = sort_sum()
+    header_string = ("Donor Name              |  Total Given | Num Gifts |"
+                     " Average Gift")
+    line = '-'*len(header_string)
+    print(header_string)
+    print(line)
+    for donor, total, num, avg in newList:
+        print(" {:<18}     $  {:>12.2f} {:>6} {:>6} {:>8.2f}".
+              format(donor, total[0], num[0], "$", avg[0]))
+    print("\n\n")
+
+
+def sort_sum():
+    # creating a new list with computed value for easy printing.
+    """
+    This function returned a sorted list of by on order amount3
+    """
+    dblist = []
+    dbSum = []
+    # quick way to convert back to list to reuse existing code.
+    # list comprehension
+    [dblist.append((i, j)) for i, j in donor_db.items()]
```

Comment:
Not the desired usage of list comprehension. No need to separately initialize `dblist` in line 73, but just write as follows:
```
dblist = [(i, j) for i, j in donor_db.items()]
```
Still, not sure why you'd want to convert the dict to a list of pairs? In fact, an easier code for that is actually: `dblist = list(donor_db.items())`.